### PR TITLE
Enable binary wheels for all python versions and OS

### PR DESCRIPTION
**Pull request by ianhbell**
I just jumped ahead and went right to scikit-build-wheels. We should test that the wheels all work, once we figure out a reasonable version of thermopack.

See https://github.com/thermotools/KineticGas/issues/28

### CODE REVIEW

**Comment by ianhbell** [comment]
Probably we should set up a time to chat about this PR. Shoot me an email at ian.bell@nist.gov and we can discuss a few things.

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+cmake_policy(VERSION 3.12)
+project(KineticGas LANGUAGES C CXX)
+
+SET(CMAKE_CXX_STANDARD 17)
+
+if(NOT MSVC)
+  # set(CMAKE_CXX_COMPILER /usr/bin/g++) # Setting these can, on some systems, cause cmake to go into an infinite loop (see: build.sh to set the compiler)
+  # set(CMAKE_C_COMPILER /usr/bin/gcc) # Setting these can, on some systems, cause cmake to go into an infinite loop (see: build.sh to set the compiler)
+  if(APPLE)
+    set(CMAKE_CXX_FLAGS "-Wall -Wextra -Wno-unknown-pragmas -Wno-sign-compare -pthread -arch arm64")
```
Comment (ianhbell) [inline CMakeLists.txt:11]: Ooops this should be fixed. The arch should not be specified this way. It should instead be something like: https://github.com/usnistgov/teqp/blob/201ceb7c59780bd93df484b90c2b2e986823116f/CMakeLists.txt#L2

**Comment by vegardjervell** [comment]
Nice stuff! I'll have a closer look at it in the coming couple of days, then we can resolve any final tweaks :)

**Comment by ianhbell** [comment]
Aside: it is better to leave the name the same in debug and release builds. Then you can just flag to build one or the other.

**Comment by vegardjervell** [comment]
I agree, what you've caught there are some more leftovers from early development that was never cleaned out. I've had a list of to-do's in cleaning up and/or streamlining quite a bit of stuff (pybind11 bindings, the integration module for collision integrals, the factorial module, etc.) which I try to find time for every now and then.

**Comment by ianhbell** [comment]
> I agree, what you've caught there are some more leftovers from early development that was never cleaned out. I've had a list of to-do's in cleaning up and/or streamlining quite a bit of stuff (pybind11 bindings, the integration module for collision integrals, the factorial module, etc.) which I try to find time for every now and then.

Yup, that's the way these things go. We'll get there. The sticking point right now is a version of thermopack that will work ok on all platforms and python versions. I guess they are not doing cibuildwheels for that either; I had volunteered to help with that some time back.

**Comment by vegardjervell** [comment]
I've had a little bit of time to look at this now, the only issue I can see is that `find_package(pybind11 REQUIRED)` doesn't appear to be very robust. I've had a hard time helping `cmake` find the package without doing a global install of `pybind11`, which I would definitely prefer to avoid. 

As of now, this package is (to my knowlege) far more commonly compiled from source for consumption rather than distribution. The way I see it, making sure it is easy for others than the lead devs to compile from source is more important than streamlining the distribution builds.

**Comment by ianhbell** [comment]
You shouldn't need to do anything magic with the python executable, it should be found properly, automatically. I know that works for me almost always and I use conda environments heavily, and I should hope the same thing would work properly with venv. For pybind11, I typically use it as a git submodule because I don't like global installs either.

Let's plan to sit down together in Boulder and go over the details.

**Comment by vegardjervell** [comment]
For some reason `cmake` insisted on using a different python version than what I was using in my venv unless I specified `set(Python_FIND_VIRTUALENV ONLY)`. I've made some modifications to make the project build for me on [this branch](https://github.com/thermotools/KineticGas/tree/new_build). I also had some issues with the binary not being included when I `pip install` the project. With `setup.py` I've just used `find_packages` to include it, but I don't have much experience with `pyproject.toml` (even though I should probably get up to speed on that anyway).

**Comment by ianhbell** [comment]
Hrm. Usually that "just works" but to be honest I don't use venv, I always use conda environments and I think the scientific python folks do too.  I would make the leap to scikit-build-core, I have been using that extensively in a new library of mine and been happy with it, allows for a very responsive build system and easy debugging.

**Comment by vegardjervell** [comment]
I've figured out of the remaining issues and put them up on #32, so I'll close this. In case anyone comes looking here I'll mention that the major problem was that files ignored by `.gitignore` were automatically ignored when the wheel was being built, so `libpykingas` needed to be explicitly included in the wheel by adding 

```
[tool.scikit-build]
sdist.include = ["libpykingas*"]
``` 

to `pyproject.toml`. Now that I've been looking a bit at this, I've noticed that theres one thing I really wish I had: A way of testing the whole system locally. Do you know if github offers any way to download the images it uses for its runners and a way to set up a local container to test the build system? I tried `act` quickly, but it didn't play very nicely with macOS on arm64.

**Comment by ianhbell** [comment]
> I've figured out of the remaining issues and put them up on #32, so I'll close this. In case anyone comes looking here I'll mention that the major problem was that files ignored by `.gitignore` were automatically ignored when the wheel was being built, so `libpykingas` needed to be explicitly included in the wheel by adding
> 
> ```
> [tool.scikit-build]
> sdist.include = ["libpykingas*"]
> ```
> 
> to `pyproject.toml`. Now that I've been looking a bit at this, I've noticed that theres one thing I really wish I had: A way of testing the whole system locally. Do you know if github offers any way to download the images it uses for its runners and a way to set up a local container to test the build system? I tried `act` quickly, but it didn't play very nicely with macOS on arm64.

I also looked into act because I had the same problem and I recall that I couldn't get it working, but I don't remember why.

**Pull request closed without merging**